### DISCUSSION TITLE
query to search API yields only 'published' cases

### DISCUSSION THREAD

**Discussion by dba-Joe** [freelawproject/courtlistener#4505] [Q&A] [answered]
Hello,

I am using the 'search' API with url pasted below.  I want the query to return cases with publication status "published," "unpublished," or "unknown." That is, I want cases that are precedential, non-precedential, and whose status is unknown.  But it is only returning cases with status "published." 

The web search interface returns cases with all three statuses, but the same search query submitted through the API returns only the published (which is, I believe, the same as "precedential") cases. 

Has anyone had similar problems? Can you suggest solutions? 

Thanks for any suggestions.

url = (f"https://www.courtlistener.com/api/rest/v4/search/?q=%22title%20vii%22%20"
       f"AND%20(%22civil%20rights%20act%22%20OR%20%2242%20U.S.C.%202000e%22%20OR%20%2242%20U.S.C.%202000%22OR%20%2242%20U.S.C.%20%C2%A7%202000e%22)%20"
       f"AND%20(employ*%20OR%20work*%20OR%20job%20OR%20discrimin*%20OR%20harass*%20"
       f"OR%20retaliat*%20OR%20promotion*)%20"
       f"AND%20(rac*%20OR%20sex*%20OR%20gender%20OR%20religio*%20OR%20%22national%20origin%22%20OR%20color)"
       f"&type=o&order_by=dateFiled%20asc&stat_Precedential=on&stat_Non-Precedential=on"
       f"&stat_Unknown%20Status=on&court={court}")

**Comment by mlissner** [comment, marked as answer]
Thanks for asking this! I think this is due to the change documented here:

https://www.courtlistener.com/help/api/rest/v4/migration-guide/#backwards-get-params

The front end is still using Solr for a little while longer, so it's still on v3, but the backend is on v4.

Try making these tweaks and I think you'll be good?

**Reply by dba-Joe** [reply]
Thanks! This worked great. Once I changed the parameters to match the language on the web page you suggested, the query worked as I wanted.
